### PR TITLE
gh: use checkout to get the downloaded data

### PR DESCRIPTION
**Pull request by isidentical**
Now, instead of caching the `data/cats_dogs` it caches the `.dvc/cache`. Also in the later step it only fetches and for both cases, it uses `dvc checkout`.

### CODE REVIEW

**Comment by efiop** [comment]
@isidentical Could you elaborate on reasoning, please?

**Comment by isidentical** [comment]
> Could you elaborate on reasoning, please?

testing something now, though the current workflow doesn't actually put cached data into `.dvc/cache` but rather just moves it to `data/cats_dogs` which is not the correct behavior. using `checkout` and caching `.dvc/cache` instead will fix this though there is a weird thing that blocks me right now, trying to test it.

**Comment by pared** [comment]
> > Could you elaborate on reasoning, please?
> 
> testing something now, though the current workflow doesn't actually put cached data into `.dvc/cache` but rather just moves it to `data/cats_dogs` which is not the correct behavior. using `checkout` and caching `.dvc/cache` instead will fix this though there is a weird thing that blocks me right now, trying to test it.

But in this change the caching has been removed completely, don't you want to cache the `.dvc/cache`?

**Comment by isidentical** [comment]
> But in this change the caching has been removed completely, don't you want to cache the .dvc/cache?

No, I was testing something. Will send to normal patch on monday

**Comment by isidentical** [comment]
The cache is working fine for linux/macos though on windows there seems to be an error :eyes: I have no idea about why, though maybe we could just disable it on windows (cc: @pared)

**Comment by pared** [comment]
@isidentical I think it might have something to do with the paths. Looking into logs, it seems like even on windows paths are provided with slashes (`C:\Windows\System32\tar.exe -z -xf D:/a/_temp/0aa681e6-55b2-4c37-a87f-dea8d0303fec/cache.tgz -P -C D:/a/dvc-bench/dvc-bench`) can't say for sure what is happening. I don't think we should be removing that. Windows and macOS builds are done to make sure we are not supporting only single system. (Even though we do not run full benchmarks on macOs and Win, we would like to get back to that at some point). 

Shouldn't pull handle existing cache? Maybe we could replace `fetch/checkout` with `pull` and cache only for macOs and Linux, and continue with investigation on windows later?

**Comment by isidentical** [comment]
@pared went that way!

**Review by pared — APPROVED** [review]

File: .github/workflows/pull_request_build.yml [hunk 1]
```diff
@@ -29,10 +29,12 @@ jobs:
...
         uses: actions/cache@v2
         with:
-          path: data/cats_dogs
-          key: ${{ hashFiles('data/cats_dogs.dvc') }}
+          path: .dvc/cache
+          key: ${{ runner.os }}-data-${{ hashFiles('**/*.dvc') }}
       - name: download data
         if: steps.cache-data.outputs.cache-hit != 'true'
+        run: dvc fetch data/cats_dogs.dvc
+      - name: checkout data
```
Comment (pared) [inline .github/workflows/pull_request_build.yml:37]: Name is not adequate to what this step is doing.
Also, now I wonder whether we need fetch step at all? If we restored the cache, what `pull` should be doing is only repopulating state, right? Maybe we just need to keep the `pull` step?

File: .github/workflows/pull_request_build.yml [hunk 2]
```diff
@@ -29,10 +29,12 @@ jobs:
         id: cache-data
         uses: actions/cache@v2
         with:
-          path: data/cats_dogs
-          key: ${{ hashFiles('data/cats_dogs.dvc') }}
+          path: .dvc/cache
+          key: ${{ runner.os }}-data-${{ hashFiles('**/*.dvc') }}
```
Comment (pared) [inline .github/workflows/pull_request_build.yml:33]: I would put here an if checking the system - now, on windows we will get additional work done

**Review by pared — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request_build.yml [hunk 3]
```diff
@@ -26,13 +26,12 @@ jobs:
...
+        if: matrix.os != 'windows-2019'
         uses: actions/cache@v2
         with:
-          path: data/cats_dogs
-          key: ${{ hashFiles('data/cats_dogs.dvc') }}
-      - name: download data
-        if: steps.cache-data.outputs.cache-hit != 'true'
+          path: .dvc/cache
+          key: ${{ runner.os }}-data-${{ hashFiles('**/*.dvc') }}
+      - name: checkout data
```
Comment (pared) [inline .github/workflows/pull_request_build.yml:34]: `pull` data :)

**Review by pared — APPROVED** [review]
One minor thingy

File: .github/workflows/pull_request_build.yml (hunk 3 above)
Comment (isidentical) [inline .github/workflows/pull_request_build.yml:34]:
```suggestion
      - name: pull data
 ```

**Review by pared — APPROVED** [review]

**Pull request merged**